### PR TITLE
Better file type checking

### LINKED ISSUE #115: Get more robust filetype identification
On https://github.com/openzim/sotoki/blob/master/sotoki/sotoki.py#L841, function `get_filetype`
We identify using headers or mine type with magic.

If we did not find something we know with headers we should fall back on magic identification (currently : return "none")

### PR DESCRIPTION
**Pull request by satyamtg**
Fixes #115

### CODE REVIEW

File: sotoki/sotoki.py [hunk 1]
```diff
@@ -840,27 +840,30 @@ def download(url, output, timeout=None):
...
-        with magic.Magic() as m:
-            mine = m.id_filename(path)
-            if "PNG" in mine:
-                ftype = "png"
-            elif "JPEG" in mine:
-                ftype = "jpeg"
-            elif "GIF" in mine:
-                ftype = "gif"
+            known_type_found = 1
+    elif (known_type_found == 0):
```
Comment (dattaz) [inline sotoki/sotoki.py]: It's should be a "if".
That way if content-type is not found in headers, it fall back to magic

Comment (satyamtg) [reply sotoki/sotoki.py]: Sorry. My bad. Such a small mistake.

**Comment by satyamtg** [comment]
> Hi,
> 
> * what's the use of `known_type_found` if we already set `ftype = "none"` ? I believe `ftype` could be used to trigger the mime behavior.
> * I think there's a typo in the code, changing `PNG` to `PDF`.
> * Also, could we compare headers using `.lower()` and get rid of the duplication?

Thanks for the review. Have made necessary changes.

**Comment by satyamtg** [comment]
> Hi,
> 
> * what's the use of `known_type_found` if we already set `ftype = "none"` ? I believe `ftype` could be used to trigger the mime behavior.
> * I think there's a typo in the code, changing `PNG` to `PDF`.
> * Also, could we compare headers using `.lower()` and get rid of the duplication?

Thanks for the review. Have made necessary changes and tested the code.

**Pull request merged**